src/components/ItemList/ItemList.tsx

```diff
@@ -0,0 +1,19 @@
+import React from "react";
+
+interface PropsType<T> {
+  items: T[];
+  renderer: (item: T, index: number, arr: T[]) => React.ReactNode;
```

Comment:
awesome awesome awesome =D

do you have any questions about why we need to pass and receive these additional 2 parameters?

Reply:
From what I've found online, it's to get over the typical Typescript error about IntrinsicAttributes. The Array properties is expected by the render function.